src/iceberg/catalog/memory/in_memory_catalog.cc

```diff
@@ -394,7 +411,32 @@ Result<std::unique_ptr<Table>> InMemoryCatalog::UpdateTable(
     const TableIdentifier& identifier,
     const std::vector<std::unique_ptr<TableRequirement>>& requirements,
     const std::vector<std::unique_ptr<TableUpdate>>& updates) {
-  return NotImplemented("update table");
+  std::unique_lock lock(mutex_);
```

Comment:
Use `std::lock_guard` instead of `std::unique_lock` if we don't need to move the guard. Anyone interested can help modify the other `std::unique_lock` in this file.